### ISSUE TITLE
Use demo.hotspot.kiwix.org as actual demo

### ISSUE THREAD

**Issue by Popolechien** [kiwix/web#117]
Currently if I'm trying to showcase our hotspot to potential partners and I don't have an image at the ready I have to go to the cardshop, order a card/config, flash it onto an SD card, put it in a Raspberry, connect to it, take a screenshot - and send that screenshot by email.

It would be good to have a few zims displayed on https://demo.hotspot.kiwix.org/ so that we can simply send people there so that they can figure it out by themselves. Typically we could have Gutenberg, TED, the same custom Wikipedia content (Geography) in French, Arabic and English, and the full Wikipedia in Spanish. I'm not sure Khan Academy could be added but that'd be another nice to have.

**Comment by rgaudin** [comment]
The demo is off (and has been in TODO for many months) due to a security issue: someone got access to the guest system and could abuse our resources resulting in our host to kill our server.

We haven't identified the security issue (might be in the demo mechanism, in the system or in any of the project) so it's not been fixed.

We could replicate some of the functionality as for the Orange Campus but it would be *similar to* and not an *actual demo*.

**Comment by Popolechien** [comment]
> We could replicate some of the functionality as for the Orange Campus but it would be similar to and not an actual demo.

Close enough.

**Comment by kelson42** [comment]
Seems to be a priority now if we want to push kiwix-hotspot and cardshop.

**Comment by benoit74** [comment]
Hi there. 
I had interest into this and I invested some time on this today.
My goal is to have a VM running in AWS with a KiwiX hotspot "simulation", so that I can easily show it to foo and friends.
I have a working "base version" based on kiwix-hotspot repo + custom tags for ansible to not apply some steps + custom ansible-playbook command + very hackish script to create a custom home page + download the ZIM files on the VM.
Still need some time to polish extremely rough edges (mainly the hackish script + some terraform code to deploy the VM instead of doing it "by hand") but it's already "working".
Running it on AWS is cool because they have the Graviton processor which is cheap and ARM-based, so quite close to the Raspberry indeed.
I can definitely transfer it to you if you have interest.

**Comment by kelson42** [comment]
@benoit74 Thx for your message. Yes, we still want to bring back the live online demo. I'm surprised it works on AWS ARM, because we tried on Scaleway and failed, but I don't remember the details. Scaleway seems to have retired its ARM based servers meanwhile, see https://news.ycombinator.com/item?id=22865925. @rgaudin will come back to you in the next days. On my side I would like to know what kind of cost is to be expected for such an instance on AWS?

**Comment by benoit74** [comment]
Cost is almost divided in three parts :
- the VM itself, about 0.01€/hour, i.e. 7.3€/month
- the storage, 0.1€/GB/month
- the outgoing traffic, first GB free then at 0.09$/GB (and this is definitely where Scaleway would have an advantage since this is free)
So the biggest part of the cost might be the storage (TED could be quite big for instance if we include all talks). Do you have any hints on what size you expect?
Regarding the outgoing traffic cost, it is a bit embarrassing this someone might use the demo "just for fun" and it might cost quite a lot.
I wonder wether AWS is the best choice. I choose it just for convenience because it is the platform I know the most, but I'm wondering wether Scaleway wouldn't be a better choice. I will give it a try tomorrow since Kiwix-serve is also supposed to work on amd64 platforms, isn't it?

**Comment by benoit74** [comment]
I confirm, it can work as well on a Scaleway DEV-1S VM, so the costs would be :
- the VM itself, 0.01$/hour, i.e. 7.3$/month
- the storage (if the 20GB of Local Storage is not sufficient) : 0.06/GB/month (assuming the VM would always be up)

**Comment by rgaudin** [comment]
Thank you @benoit74 for all those valuable information. Last demo was 128GB so that would take us around 15$/mo total on scaleway.

I can't recall why it didn't work on scaleway ARM when we tried but I have the feeling you're only setting up kiwix-serve which is not what we're looking for. We already have a kiwix-serve *demo*.

Ideally, we want something that takes a kiwix-hotspot image as input and offers its content on the web . Goal is to replicate as much as possible the real hotspot experience (once passed the captive portal).
Except for speed, I think we were quite happy with [the previous qemu-based solution](https://github.com/kiwix/kiwix-hotspot/tree/master/online-demo) (hence the attempt to make it work on ARM hardware).

Now we had to shut this demo down because someone exploited it and used either the host or the qemu system for malicious activities (that scaleway detected from traffic).
As mentioned in my comment above, we don't know what was exploited to gain access, but I doubt it was the host and would lean toward an insecure app that was used to leverage the fact we use the default password for the raspberry user (which can elevate).

Can you share more details about what you did? Do you have a URL to share for your demo prototype?

**Comment by Popolechien** [comment]
@kelson42 @rgaudin Do we know who set up http://demo.ideascube.org/en/ and how they did it?

**Comment by rgaudin** [comment]
@Popolechien this is not kiwix-hotspot, that's ideascube.

**Comment by benoit74** [comment]
@rgaudin Actually the VM is down, I will rebuild it soon and send you the link +  the git branch I used to build the VM. 

Basically I started a VM in the Cloud, removed everything from the ansible which is "hardware oriented", i.e. wifi, partitions, clock, captive-portal and let everything else install just like it does in QEMU image and let ansible run on the VM itself. To have some data, I hacked a python script which I'm running manually on the VM for now, to download content, place it where it is supposed to be and then let ansible finish the installation.

I don't see any value in using a real QEMU image in my case, just drawbacks (tons on complicated network stuff, additional security risks, ...). After all, I just want to show the interface to kiwix like it is on the hotspot. With what I've built, I even have access to EduPI, KAlite, and stuff like that, even if it is not my goal for now. So if you still want to use the QEMU method, then my attempt will indeed be useless for you.

**Comment by rgaudin** [comment]
No, qemu is not a requirement at all. Was just convenient as it allowed creating an almost identical setup.

What you've created sounds great. We can tweak it so it does all this from a kiwix-hotspot config and voila. As you suggested, filtering all those ansible tasks via an `hardware` tag would allow us to get that into the repo.

Only remaining concern would be the security issue.

**Comment by benoit74** [comment]
Hello again,

I just realized that indeed, due to my config, only kiwix-serve was installed, the links on the home page were generated but nothing was installed nor working ... Sorry about that false hope ! ;-)

I tried to install other software stacks but I realized that it was not working and would be very hard since it rapidly became obvious that the hotspot functioning relies on a lot of specificities for those additional packages (specific node version for ARM downloaded manually, custom venv for the user 'user' which I previously removed since I believed it was used only for 'clock', ...). I came to the conclusion that (at least for now given the current state of the code), it is not reliable to make a fresh install of all components. Especially given the fact that all new components would involve a new adaptation of the code base. Your previous QEMU-based solution is probably more realistic on this matter, since it would run exactly the same software stack without any need to maintain side-by-side compatibility for hardware and not_hardware compatibility.

Regarding the security issue, I could imagine that it is involved by one of the multiple packages installed.

I don't know how / where we could move forward. Maybe looking at how ideascube is achieving to run its demo?
